nodejs/publish-layers.sh

```diff
@@ -113,6 +115,54 @@ function publish-nodejs16x-x86 {
     done
 }
 
+function build-nodejs18x-arm64 {
+	echo "Building new relic layer for nodejs18.x (arm64)"
+	rm -rf $BUILD_DIR $NJS18X_DIST_ARM64
+	mkdir -p $DIST_DIR
+	npm install --prefix $BUILD_DIR newrelic@latest
+	mkdir -p $BUILD_DIR/node_modules/newrelic-lambda-wrapper
+	cp index.js $BUILD_DIR/node_modules/newrelic-lambda-wrapper
+	download_extension arm64
+	zip -rq $NJS18X_DIST_ARM64 $BUILD_DIR $EXTENSION_DIST_DIR $EXTENSION_DIST_PREVIEW_FILE
+	rm -rf $BUILD_DIR $EXTENSION_DIST_DIR $EXTENSION_DIST_PREVIEW_FILE
+	echo "Build complete: ${NJS18X_DIST_ARM64}"
+}
+
+function build-nodejs18x-x86 {
+	echo "Building new relic layer for nodejs18.x (x86_64)"
+	rm -rf $BUILD_DIR $NJS18X_DIST_X86_64
+	mkdir -p $DIST_DIR
+	npm install --prefix $BUILD_DIR newrelic@latest
+	mkdir -p $BUILD_DIR/node_modules/newrelic-lambda-wrapper
+	cp index.js $BUILD_DIR/node_modules/newrelic-lambda-wrapper
+	download_extension x86_64
+	zip -rq $NJS18X_DIST_X86_64 $BUILD_DIR $EXTENSION_DIST_DIR $EXTENSION_DIST_PREVIEW_FILE
+	rm -rf $BUILD_DIR $EXTENSION_DIST_DIR $EXTENSION_DIST_PREVIEW_FILE
+	echo "Build complete: ${NJS18X_DIST_X86_64}"
+}
+
+function publish-nodejs18x-arm64 {
+    if [ ! -f $NJS18X_DIST_ARM64 ]; then
+      echo "Package not found: ${NJS18X_DIST_ARM64}"
+      exit 1
+    fi
+
+    for region in "${REGIONS_ARM[@]}"; do
+      publish_layer $NJS18X_DIST_ARM64 $region nodejs18.x arm64
+    done
+}
+
+function publish-nodejs18x-x86 {
+    if [ ! -f $NJS18X_DIST_X86_64 ]; then
+      echo "Package not found: ${NJS18X_DIST_X86_64}"
+      exit 1
+    fi
+
+    for region in "${REGIONS_X86[@]}"; do
+      publish_layer $NJS18X_DIST_X86_64 $region nodejs18.x x86_64
+    done
+}
+
```

Comment:
Should this be a little less copy-pastey?

Reply:
It arguably could be more DRY, though in its current form, adding/removing runtime versions is fairly explicit, which can be an advantage.